### ISSUE TITLE
Nox-poetry outputs bad build command on Python 3.11

### ISSUE THREAD

**Issue by drhagen** [cjolowicz/nox-poetry#941]
Python 3.11 changed [how enums are printed](https://docs.python.org/3.11/library/enum.html#enum.Enum.__str__), and as a result,  nox-poetry fails on Python 3.11. [This `--format` line](https://github.com/cjolowicz/nox-poetry/blob/main/src/nox_poetry/poetry.py#L140) contains the problem:

```python
        output = self.session.run_always(
            "poetry",
            "build",
            f"--format={format}", # should be f"--format={format.value}"
            "--no-ansi",
            external=True,
            silent=True,
            stderr=None,
        )
```

See [this CI run](https://github.com/drhagen/serialite/actions/runs/3320818954/jobs/5487654727) for an example of nox-poetry outputting a weird command and crashing.

Note that you could use the [new `StrEnum`](https://docs.python.org/3.11/library/enum.html#enum.StrEnum) in place of inheriting from `str, Enum`, but then you would need version guards around the `DistributionFormat` class definition, which may not be worth it.

**Comment by johnthagen** [comment]
This (incompatible) change is intentional and documented in the What's New:

- https://docs.python.org/3/whatsnew/3.11.html#enum

> Changed [Enum.__format__()](https://docs.python.org/3/library/enum.html#enum.Enum.__format__) (the default for [format()](https://docs.python.org/3/library/functions.html#format), [str.format()](https://docs.python.org/3/library/stdtypes.html#str.format) and [f-string](https://docs.python.org/3/glossary.html#term-f-string)s) of enums with mixed-in types (e.g. [int](https://docs.python.org/3/library/functions.html#int), [str](https://docs.python.org/3/library/stdtypes.html#str)) to also include the class name in the output, not just the member’s key. This matches the existing behavior of [enum.Enum.__str__()](https://docs.python.org/3/library/enum.html#enum.Enum.__str__), returning e.g. 'AnEnum.MEMBER' for an enum AnEnum(str, Enum) instead of just 'MEMBER'.

**Comment by rvodden** [comment]
@cjolowicz could this be merged please? It's blocking the use of 3.11 with nox-poetry :-)
